src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,185 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (array.length > size) {
+            addElement(value);
+        } else {
+            grow();
+            addElement(value);
+        }
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        checkIfIndexNegative(index);
+        checkIfIndexPresent(index);
+        if (array.length > size) {
+            addElement(value, index);
+        } else {
+            grow();
+            addElement(value, index);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int newSize = size + list.size();
+        if (newSize < array.length) {
```

Comment:
Don't begin method implementation with an empty line. Remove the empty line at the beginning of the method.